api/users/utils/mailer_lite.py

```diff
@@ -0,0 +1,99 @@
+import json
+import typing
+from contextlib import AbstractContextManager
+
+import requests
+from django.conf import settings
+
+from users import models
+from util.util import postpone
+
+MAX_BATCH_SIZE = 50
+
+
+class MailerLiteBaseClient:
+    resource = None
+
+    def __init__(self):
+        self.base_url = settings.MAILERLITE_BASE_URL
+
+    def _post(self, data):
+        url = self.base_url + self.resource
+        requests.post(url, data=json.dumps(data), headers=self._get_request_headers())
+
+    @staticmethod
+    def _get_request_headers() -> typing.Mapping:
+        return {
+            "X-MailerLite-ApiKey": settings.MAILERLITE_API_KEY,
+            "Content-Type": "application/json",
+        }
+
+
+class MailerLite(MailerLiteBaseClient):
+    resource = "subscribers"
+
+    @postpone
+    def subscribe(self, user: "models.FFAdminUser"):
+        self._subscribe(user)
+
+    @postpone
+    def subscribe_organisation(self, organisation_id: int):
+        return self._subscribe_organisation(organisation_id)
+
+    def _subscribe(self, user: "models.FFAdminUser"):
+        if not user.is_subscribed:
```

Comment:
This seems a bit counterintuitive. I think probably because the `is_subscribed` flag is a bit misleading. Perhaps something like `has_agreed_to_marketing` would be more appropriate?